04-ruby-mvc-sinatra/object_oriented_programming_in_ruby/1_classes_and_objects.md

```diff
@@ -16,6 +16,10 @@
 Object-oriented programming is the idea that our programs consist mainly of
 objects, which have attributes (aka properties), and methods (aka behavior).
 
+Why do we need objects? Let's take a look at [amazon](http://www.amazon.com/).
+
+Let's search for speakers. When we hit enter we can see a whole bunch of different speakers. Did the people that coded amazon hard code all of these speakers? Think of each different speaker as an object. These objects have properties like a title, url, price, or rating. We need to be able to store and consolidate information into objects so that we can access/display them on a webpage.
+
```

Comment:
classic example is a car. you can drive a car without knowing how the engine works. The hood encapsulates the internal processes.